natiami/assignment3/FindWordsTest.cpp

```diff
@@ -0,0 +1,51 @@
+#include "FindWordsTest.h"
+
+void FindWordsTest::test(){
+	test_findAllWords();
+	test_EmptyGrid();
+	test_EmptyDictionary();
+}
+
+Grid* FindWordsTest::initGrid(){

```

Comment:
Tip 1:
This can be replaced with a one liner, using vector initializer list:
```C++
return new Grid{{'A', 'A', 'R'}, {'T', 'C', 'D'}};
```
Explaination:
A vector can be initialized with a list using braces initialization, for example: `vector<int> v{1,2,3};`
Here you have a vector of vectors, so each element of the vector is a vector itself, so you can use nested braces.

-----
Tip 2:
If you dynamically allogate a grid (using `new` operator) and return it by pointer (`grid*`), then the caller should release it:
```C++
Grid* grid = initGrid();
// Use grid...
delete grid;
```
or there will be a memory leak!

So here you can return the Grid by value, so there is no need to deal with memory management:
```C++
Grid initGrid() { return Grid{{'A', 'A', 'R'}, {'T', 'C', 'D'}}; }
```
Or even:
```C++
Grid initGrid() { return {{'A', 'A', 'R'}, {'T', 'C', 'D'}}; }
```

Another option, is to make `initGrid` return nothing, and initialize the `grid` field of the test class:
```C++
void initGrid() {
    grid = Grid{{'A', 'A', 'R'}, {'T', 'C', 'D'}};  // grid is a field of the test class.
    // Or:
    grid = {{'A', 'A', 'R'}, {'T', 'C', 'D'}};
}
```